src/utils.rs

```diff
@@ -253,8 +253,10 @@ pub fn parse_batch(msg:&Batch)-> Ruddermessage{
     for i in &msg.batch {
         match i {
             BatchMessage::Identify(a_) =>{
-                let mut final_context: Value = a_.context.clone().unwrap_or(json!({}));
-                merge(&mut final_context, modified_context.clone() );
+                let mut event_context: Value = a_.context.clone().unwrap_or(json!({}));
+                let mut final_context: Value = batch_context.clone();
+                merge(&mut event_context, default_context.clone());
```

Comment:
Are we not allowing default context to be overridden now?

Reply:
Sounds good. You've also fixed another bug in the previous implementation.

Reply:
However, you're not solving this for the batch-level context. See my other comment.